namer/core/src/main/scala/io/buoyant/namer/RewritingNamerInitializer.scala

```diff
@@ -0,0 +1,24 @@
+package io.buoyant.namer
+
+import com.fasterxml.jackson.annotation.JsonIgnore
+import com.twitter.finagle.{Namer, Path}
+import com.twitter.finagle.Stack.Params
+import io.buoyant.namer.util.PathMatcher
+
+class RewritingNamerInitializer extends NamerInitializer {
+  override val configId = "io.l5d.rewrite"
+  val configClass = classOf[RewritingNamerConfig]
+}
+
+case class RewritingNamerConfig(
+  pattern: String,
+  name: String
+) extends NamerConfig {
+  assert(pattern != null)
+  assert(name != null)
+
+  @JsonIgnore
+  override def defaultPrefix: Path = Path.read("/io.l5d.rewrite")
```

Comment:
I sort of feel that we should require explicit prefixes for this namer?  Since it seems awkward to ever have `/#/io.l5d.rewrite` be meaningful...